### DISCUSSION TITLE
use DVB-SI to mark service(s) as unavailable

### DISCUSSION THREAD

**Discussion by BartmanEH** [EricBerendsen/dvbinspector#30] [General]
Hello Eric,

Looking for some help from a DVB-SI expert. I'm wondering if there's a way to temporarily modify DVB-SI information to indicate that a subset of services/channels are temporarily unavailable.

The use case is as follows: for an incoming MPTS with a large number of services, I would like to forward only a subset of services but I would like the SI to indicate (advertise, if you will) the full gamut of potential services in the original MPTS. Through another mechanism, a different subset of services may be chosen and the MPTS manipulated in such a way to 'mark' the newly non-chosen services as 'unavailable' and correspondingly mark the desired services as available again (restore normal DVB-SI related information).

Essentially, cull the MPTS to only a subset of desired services but keep valid SI showing all possible original services, just modified in such a way that the resulting culled MPTS is still valid but missing PID data for the non-subset services.

Thanks for your help and guidance.

Bart

**Comment by EricBerendsen** [comment]
Hi Bart,

I would not call myself a DVB-SI expert, I just created DVB Inspector to learn about DVB. I am not working in the industry. 

Still can try to give you an answer. You could set the running_status field in the SDT of the services you want tot remove to 3
(pausing) or 5 (service off-air), instead of 4 running. You can then remove the audio/video PIDs, PMTs and entries in the PAT for the removed services.  Some more info in [Chapter 4.3 of ETSI TS 101 211 V1.13.1  Implementation and usage of Service Information](https://www.etsi.org/deliver/etsi_ts/101200_101299/101211/01.13.01_60/ts_101211v011301p.pdf)

Not sure how/if 'removed' services will be shown at consumer end. Think you will need a lot of testing and experimenting with this dynamic setup. Sounds interesting!

**Comment by BartmanEH** [comment]
Thank you, Eric. Much appreciated for pointing us in the right direction.

Our device will be at the end point in this case so we can control its reaction, however, in some cases there may be other downstream equipment such as a multiplexer. In this case we would like the TS to be valid and not produce "no PID data" alarms, for example.

We will investigate the running_status field in the SDT in the hopes that it accomplishes what we need--a valid TS, but at reduced bandwidth, with only the PID data we need at any given time, along with all the original SI advertising what services are available to be requested.

Sounds promising!
